# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/generic/ViewportFrame.cpp
# Commit: fdee5a641aee
# Full Hash: fdee5a641aee35ea342ca6c0827dc52fce2b7ec7
# Author: Matt Woodrow <mwoodrow@mozilla.com>
# Date: 2021-05-19 08:32:22
# Regressor Bug: 1540737
# File Overlap Count: 2
# Description:
#   Bug 1540737 - Don't allow constructing an AutoBuildingDisplayList with a new frame without requiring explicit rects in the new coordinate space. r=miko
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D114675
# ==============================================================================

diff -r b091009018d3 -r fdee5a641aee layout/generic/ViewportFrame.cpp
--- a/layout/generic/ViewportFrame.cpp	Wed May 19 01:11:31 2021 +0000
+++ b/layout/generic/ViewportFrame.cpp	Wed May 19 01:11:32 2021 +0000
@@ -229,8 +229,10 @@
   if (topLayerList.IsEmpty()) {
     return nullptr;
   }
-  nsDisplayListBuilder::AutoBuildingDisplayList buildingDisplayList(aBuilder,
-                                                                    this);
+  nsPoint offset = aBuilder->GetCurrentFrame()->GetOffsetTo(this);
+  nsDisplayListBuilder::AutoBuildingDisplayList buildingDisplayList(
+      aBuilder, this, aBuilder->GetVisibleRect() + offset,
+      aBuilder->GetDirtyRect() + offset);
   // Wrap the whole top layer in a single item with maximum z-index,
   // and append it at the very end, so that it stays at the topmost.
   nsDisplayWrapList* wrapList = MakeDisplayItemWithIndex<nsDisplayWrapList>(